src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1411,7 +1411,7 @@
     <value>For determining whether the object contains or not any items, prefer using 'IsEmpty' property rather than retrieving the number of items from the 'Count' property and comparing it to 0 or 1.</value>
   </data>
   <data name="PreferIsEmptyOverCountMessage" xml:space="preserve">
-    <value>Prefer 'IsEmpty' over 'Count' to determine whether the object contains or not any items</value>
+    <value>Prefer 'IsEmpty' over 'Count' to determine whether the object is empty</value>
```

Comment:
I think you need to mention two alternatives to properly use "whether".

```suggestion
    <value>Prefer 'IsEmpty' over 'Count' to determine whether the object is empty or not</value>
```

```suggestion
    <value>Prefer 'IsEmpty' over 'Count' to determine whether or not the object is empty</value>
```

We could even rephrase to not use the word
```suggestion
    <value>Prefer 'IsEmpty' over 'Count' to determine if the object is empty</value>
```

Reply:
@Jozkee
I'm not a native speaker so I looked it up.,The first two resources I found don't back that up
( https://english.stackexchange.com/a/48773/398754, http://www.youtube.com/watch?v=Klnroe1UBRs&t=1m23s <-- here making a specific example with whether and stating that "or not"  is not necessary).
So I think it's rather a stylistic choice. Personally I would go for the shorter version unless there's ambiguity.
That would make me lean toward the `if` version (probably also most commonly understood by non native speakers; every programmer will know `if` introduces a condition). Though the guy in the youtube video said that `whether` specifies a binary condition and `if` goes well with multiple options.

Maybe we could have an odd number of native speakers vote on a version? :D

Reply:
Lets go with the changes and if required/based on feedback, we can change them in future.